test/utils/refresh.go

```diff
@@ -0,0 +1,92 @@
+package utils
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/require"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+type Refresh struct {
+	TestRefreshServicesAndConfigPaths bool
+}
+
+func TestRefresh(t *testing.T, snapName, serviceName string, conf Refresh) {
+	t.Run("refresh", func(t *testing.T) {
+		if conf.TestRefreshServicesAndConfigPaths {
+			testRefresh(t, snapName, serviceName)
+		}
+	})
+}
+
+func testRefresh(t *testing.T, snapName, serviceName string) {
+	const refreshChannel = "latest/beta"
+	var refreshRevision string
+
+	t.Cleanup(func() {
+		if LocalSnap != "" {
+			SnapRemove(t, snapName)
+			SnapInstallFromFile(t, LocalSnap)
+		} else {
+			SnapRemove(t, snapName)
+			SnapInstallFromStore(t, snapName, ServiceChannel)
+		}
+		WaitPlatformOnline(t)
+	})
+
+	originalVersion := SnapVersion(t, snapName)
+	originalRevision := SnapRevision(t, serviceName)
+
+	t.Run("refresh services", func(t *testing.T) {
+		SnapRefresh(t, snapName, refreshChannel)
+		refreshVersion := SnapVersion(t, snapName)
+		WaitPlatformOnline(t)
+
+		t.Logf(`Successfully upgraded:
+		from: %s
+		to:   %s`,
+			originalVersion, refreshVersion)
```

Comment:
How about:
```suggestion
		t.Logf("Successfully upgraded:\n\t from: %s\n\t to: %s", originalVersion, refreshVersion)
```

Reply:
I meant "Successfully upgraded from version"

If you wanna include both version and revision (useful when testing another build of the same version), then:
> "Successfully upgraded from vX.Y.Z (revR) to vX.Y.Z (revR)"

Reply:
Thanks, I will go with printing both version and revision as the following config pat test will use revision.